### ISSUE TITLE
"zfs diff" is unable to determine path or stats for an object

### ISSUE THREAD

**Issue by chungy** [openzfs/zfs#948]
> (OS info: Arch Linux x86_64 with stock kernel 3.5.3 and SPL/ZFS 0.6.0_rc10)

I have a filesystem I'm using to host a CentOS 5 container within, but when I try to use `zfs diff` I get an error; I have performed a `zpool scrub` that returned precisely zero errors in the pool, but it did not resolve this situation.  No messages appear in dmesg about it, either.

Example run:

```
# zfs list -rt all philipjfry/lxc/centos5
NAME                           USED  AVAIL  REFER  MOUNTPOINT
philipjfry/lxc/centos5        1.53G  68.4G  1.52G  /lxc/centos5
philipjfry/lxc/centos5@fresh  1.09M      -  1.52G  -
# zfs diff philipjfry/lxc/centos5@fresh
M       /lxc/centos5/
M       /lxc/centos5/var/run
Unable to determine path or stats for object 7734 in philipjfry/lxc/centos5@fresh: No such file or directory
```

Sorry for the quite possible lack of information to reproduce it (short of myself running it), but I'll be happy to respond with more info if I'm asked to (and how).

**Comment by Bernard-Limogne** [comment]
If you create a ciphered dataset  with 'dnodesize=auto', the result of the command 'zfs diff' is quite wrong after any update of the dataset .

**Comment by behlendorf** [comment]
@Bernard-Limogne the issue you're describing sounds like the issue addressed by #9343.  The fix for this is queued up for 0.8.3.
